src/components/NewBookingModal/NewBooking/form.js

```diff
@@ -129,7 +129,17 @@ class RawForm extends Component {
         <DatePicker
           id="startDate"
           selected={values.startDate}
-          onChange={value => setFieldValue('startDate', value)}
+          onChange={value => {
```

Comment:
Personally I'd separate this out into a separate function. At a glance something like 

dateSelectionChanged = (selectedValue) => {
     if (values.endDate.isBefore(value)) {
              setFieldValue('startDate', value);
              setFieldValue('endDate', value);
            } else if (values.isHalfDay) {
              setFieldValue('endDate', value);
              setFieldValue('startDate', value);
            } else {
              setFieldValue('startDate', value);
            }
}

Althought even better could be to go down the container/component approach.

Reply:
If it's extracted into its own method you'll have to pass the setFieldValue prop further down as a property when you call it. This means the extracted method would be coupled to forms, which means I don't think it should be extracted personally.